acceptance/tests/resource/file/ticket_6448_file_with_utf8_source.rb

```diff
@@ -0,0 +1,82 @@
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+test_name 'Ensure a file resource can have a UTF-8 source attribute, content, and path when served via a module' do
+  skip_test 'requires a master for serving module content' if master.nil?
+
+  tmp_environment = mk_tmp_environment_with_teardown(master, File.basename(__FILE__, '.*'))
+  agent_tmp_dirs = {}
+
+  agents.each do |agent|
+    agent_tmp_dirs[agent.hostname] = agent.tmpdir(tmp_environment)
+  end
+
+  teardown do
+    step 'remove all test files on agents' do
+      agents.each { |agent| on(agent, "rm #{agent_tmp_dirs[agent.hostname]}", :accept_all_exit_codes => true) }
+    end
+    # note - master teardown is registered by #mk_tmp_environment_with_teardown
+  end
+
+  step 'create unicode source file served via module on master' do
+    # 静 \u9759 0xE9 0x9D 0x99 http://www.fileformat.info/info/unicode/char/9759/index.htm
+    # 的 \u7684 0xE7 0x9A 0x84 http://www.fileformat.info/info/unicode/char/7684/index.htm
+    # ☃ \2603 0xE2 0x98 0x83 http://www.fileformat.info/info/unicode/char/2603/index.htm
+    setup_module_on_master = <<-MASTER_MANIFEST
+      File {
+        ensure => directory,
+        mode => "0755",
+      }
+
+      file {
+        '#{environmentpath}/#{tmp_environment}/modules/utf8_file_module':;
+        '#{environmentpath}/#{tmp_environment}/modules/utf8_file_module/files':;
+      }
+      
+      file { '#{environmentpath}/#{tmp_environment}/modules/utf8_file_module/files/\u9759\u7684':
+        ensure => file,
+        content => "\u2603"
+      }
+    MASTER_MANIFEST
+    apply_manifest_on(master, setup_module_on_master, :expect_changes => true)
```

Comment:
Is there not already a helper somewhere to do this / should we just provide some fixture files in the repo that get copied over?

Because Puppet is being used to create the files, it's (remotely) possible that Puppet misbehaves in the same way on both sides of the wire, and gets a false positive success.

There used to be a PMT test at `acceptance/tests/modules/install/long_path_name.rb` from https://github.com/puppetlabs/puppet/pull/4892/files that installed a real module.


I can go either way on this one - testing via manifest on the master could also help detect problems with writing utf-8 files / paths.